src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -12,6 +21,107 @@ public CollectionSortClauseBuilder(CollectionSchemaObject collection) {
 
   @Override
   public SortClause buildAndValidate(ObjectNode sortNode) {
+    // $lexical is only special for Collections: handle first
+    JsonNode lexicalValue = sortNode.get(DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
+    if (lexicalValue != null) {
+      // We can also check if lexical sort supported by the collection:
+      if (!schema.lexicalConfig().enabled()) {
+        throw ErrorCodeV1.LEXICAL_NOT_ENABLED_FOR_COLLECTION.toApiException(
+            "Lexical search is not enabled for collection '%s'", schema.name());
+      }
+      if (sortNode.size() > 1) {
```

Comment:
buildAndValidateExpression gets the totalFields count and does checks like this for $vectorize, unclear why the lexical is handled differently